### ISSUE TITLE
[HOLD for payment 2023-08-03] [$1000] Web - Temporary display of welcome message after deleting the last message

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22591]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login to the  userA’s account
2. Go to the Setting > Workspace > New Workspace
3. Navigate to the "New room" section
4. Enter a name for the room and select the created workspace
5. Set the visibility of the room to "Public"
6. Click on the "Create room" button
7. Go to the room details > Settings > Welcome message
8. Enter welcome message and save it
9. Now go to Share code > Copy URL to clipboard
10. Open another browser and log in with userB account
11. Join the room by pasting the copied room URL
12. Pin the room
13. Send a message after joining the room and delete it

## Expected Result:
The welcome message should not be displayed for a while after deleting the last message

## Actual Result:
The welcome message is displayed for a while after deleting the last message

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.38-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/321bd9db-9f2d-424e-9901-cd3771abf400

https://github.com/Expensify/App/assets/93399543/f949697f-00f7-430f-b6c4-a24895037c03

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ayazhussain79
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689001566882329

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ddbab3fa4cdb7228</li>
        <li>Upwork Job ID: 1679926471892664320</li>
        2023-07-14<li>Automatic offers: </li>
<ul>
<li> |  | 0</li>
<li>tienifr | Contributor | 25679418</li>
<li> |  | 0</li>
</ul></li>
    </ul>
</details>

**Comment by mallenexpensify** [comment]
Added this to the op  @ayazhussain79, otherwise the room disappears from LHN. 
> 12. Pin the room

If you don't pin it, the room disappears from LHN, I'm unsure how someone would find it once it's gone if they don't know the room name.  I wonder if can _easily_ fix that as part of this bug or if we should have two bug issues?  Or... maybe the one I found is a feature request? I'm uncertain of expected behaviour.   What do you think @allroundexperts ?

**Comment by s-alves10** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When deleting the last message, welcome message is displayed temporarily.

### What is the root cause of that problem?

We calculate the `lastMessageText`(this is displayed in LHN) here
https://github.com/Expensify/App/blob/e2e4057ce73b1d0b3762315aed3a1896d48c018a/src/libs/actions/Report.js#L902

As you can see, we get the last visible message from the report actions. That's why last visible action's message will be displayed in the LHN regardless it is whisper or not(Remember the welcome message is also a visible action)

This is the root cause

### What changes do you think we should make in order to solve the problem?

We need to filter out whisper actions when getting `lastMessageText`.

Add the 3rd parameter to the `ReportActionsUtils.getLastVisibleMessage` and `ReportActionsUtils.getLastVisibleAction` functions named `includeWhisper` and change the logic for getting visible actions as follows.

```
function getLastVisibleAction(reportID, actionsToMerge = {}, includeWhisper = true) {
    ...
    const visibleActions = _.filter(actions, (action) => !isDeletedAction(action) && (includeWhisper || !isWhisperAction(action)));


    ...
}

function getLastVisibleMessage(reportID, actionsToMerge = {}, includeWhisper = true) {
    const lastVisibleAction = getLastVisibleAction(reportID, actionsToMerge, includeWhisper);

    ...
}
```

This works as expected.

<details>
    <summary>Result</summary>


</details>

### What alternative solutions did you explore? (Optional)

**Comment by mallenexpensify** [comment]
Thanks @allroundexperts , and.... if 'room not showing in LHN' is a different/separate issue, can you think of any reason we'd want to address that first, before this bug report?  I don't think it's necessary but would like bonus 👀  in case I'm missing something

**Comment by mallenexpensify** [comment]
Proposed the new feature https://expensify.slack.com/archives/C01GTK53T8Q/p1689714154006189
👀 please @allroundexperts to ensure I didn't miss anything

**Comment by mallenexpensify** [comment]
We're not going to address this issue of the room _disappearing_ in LHN because, once some/anyone posts in the room, it'll show in LHN, so it's safe to assume that people will post in room and, if not, then the room isn't really _useful_.   

So... we're clear to move forward with this specific issue. @allroundexperts , will you please review the proposals above and provide feedback? Thx

**Comment by stitesExpensify** [comment]
@tienifr proposal looks good to me!

**Comment by stitesExpensify** [comment]
LGTM @allroundexperts. Can you fill out the checklist and then we can get this paid out?

**Comment by stitesExpensify** [comment]
You can't check boxes!? Then why do we tag you on each of them haha

Either way, let's get this taken care of if we can @mallenexpensify

**Comment by ayazhussain79** [comment]
@mallenexpensify offer accepted, Thank you

**Comment by JmillsExpensify** [comment]
Reviewed the details for @allroundexperts. $1,500 approved for payment in NewDot based on the BZ summary above.

**Comment by stitesExpensify** [comment]
Is this ready to be closed?

**Comment by ayazhussain79** [comment]
Payment pending on Upwork

**Comment by mallenexpensify** [comment]
Issue reporter: @ayazhussain79 paid $250 via Upwork
Contributor: @tienifr paid $1500 via Upwork
Contributor+: @allroundexperts is due $1500, paid outside of Upwork.

Testrail update GH - https://github.com/Expensify/Expensify/issues/312295
